File: site/page/clients/download-term.pl
#!/usr/bin/perl
use strict;
use warnings;

use LWP::UserAgent;
use XML::XPath;
use XML::XPath::XMLParser;

my $ua = LWP::UserAgent->new;
my $req = HTTP::Request->new(GET => 'http://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0003824&format=oboxml');
my $res = $ua->request($req, "term.xml");

my $xpath = XML::XPath->new(filename => "term.xml");

my $nodeset = $xpath->find("/obo/term/name");
foreach my $node ($nodeset->get_nodelist) {
  print XML::XPath::XMLParser::as_string($node) . "\n";
}

exit;

File: site/page/clients/DownloadTerm.java
import org.w3c.dom.*;

import javax.xml.parsers.*;
import javax.xml.xpath.*;
import java.io.*;
import java.net.*;


/**
 * Sample java client to demonstrate downloading information about a GO Term
 *
 */

public class DownloadTerm {
    public static void main(String[] args) throws Exception {
        // URL a GO Term in OBO xml format
        URL u=new URL("http://www.ebi.ac.uk/QuickGO/GTerm?id=GO:0003824&format=oboxml");
        // Connect
        HttpURLConnection urlConnection = (HttpURLConnection) u.openConnection();

        // Parse an XML document from the connection
        InputStream inputStream = urlConnection.getInputStream();
        Document xml=DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(inputStream);
        inputStream.close();

        // XPath is here used to locate parts of an XML document
        XPath xpath=XPathFactory.newInstance().newXPath();

        //Locate the term name and print it out
        System.out.println("Term name:"+xpath.compile("/obo/term/name").evaluate(xml));

    }
}
